src/azdata.proposed.d.ts

```diff
@@ -413,6 +413,24 @@ declare module 'azdata' {
 		azurePortalEndpoint?: string;
 	}
 
+	/**
+	 * An interface for receiving diagnostic data from a provider.
+	 */
+	export interface Diagnostics {
+		handleErrorCode(errorCode: number, errorMessage: string): Thenable<diagnostics.ErrorDiagnosticsResponse>;
+	}
+
+	export namespace diagnostics {
+		export interface ErrorDiagnosticsResponse {
+			errorAction: string;
```

Comment:
Can this be an enum instead, with 0 pointing to default action, i.e. to show error dialog.